anta/tests/interfaces.py

```diff
@@ -597,3 +598,78 @@ def test(self) -> None:
             self.result.is_failure(f"IP virtual router MAC address `{self.inputs.mac_address}` is not configured.")
         else:
             self.result.is_success()
+
+
+class VerifyInterfacesSpeed(AntaTest):
+    """
+    Verifies the speed, lanes, auto-negotiation status, and mode as full duplex for interfaces.
+    - If speed is auto then verify auto-negotiation as success and mode as full duplex.
+    - If speed is auto with a value (auto 10g) then verify auto-negotiation as success, mode as full duplex and speed/lanes as per input.
+    - If speed is forced with a value (forced 10g) then verify mode as full duplex and speed as per input.
+    - If speed with lane (100g-8) then verify mode as full duplex and speed/lanes as per input.
+
+    Expected Results:
+        * Success: The test will pass if an interface is configured with the correct speed, lanes, auto-negotiation and mode as full duplex.
+        * Failure: The test will fail if an interface is not found, the speed or lanes do not match with input,
+                   auto-negotiation is not correct or mode is not full duplex.
+    """
+
+    name = "VerifyInterfacesSpeed"
+    description = "Verifies the speed, lanes, auto-negotiation status, and mode as full duplex for interfaces."
+    categories = ["interfaces"]
+    commands = [AntaCommand(command="show interfaces")]
+
+    class Input(AntaTest.Input):
+        """Inputs for the VerifyInterfacesSpeed test."""
+
+        interfaces: List[InterfaceDetail]
+        """List of interfaces to be tested"""
+
+        class InterfaceDetail(BaseModel):
+            """Detail of an interface"""
+
+            name: EthernetInterface
+            """Name of the interface"""
+            speed: InterfaceSpeedPattern
+            """Speed of an interface in Gigabits per second"""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        self.result.is_success()
+        command_output = self.instance_commands[0].json_output
+
+        # Iterate over all the interfaces
+        for interface in self.inputs.interfaces:
+            intf = interface.name
+
+            # Check if interface exists
+            if not (interface_output := get_value(command_output, f"interfaces.{intf}")):
+                self.result.is_failure(f"Interface `{intf}` not found.")
+                continue
+
+            auto_negotiation = interface_output.get("autoNegotiate")
+            duplex = interface_output.get("duplex")
+            actual_speed = interface_output.get("bandwidth")
+            actual_lanes = interface_output.get("lanes")
+            speed, lanes = extract_speed_and_lane(interface.speed)
+
+            # Collecting actual interface details
+            actual_interface_output = {
+                "auto negotiation": auto_negotiation if "auto" in interface.speed else "N/A",
```

Comment:
the N/A string looks weird and a bit forced- could we not have the nested function handle missing keys instead and we use `None`?

Reply:
updated the util function get_failed_logs to handle None object.

actual_data = actual_output.get(element, "not found")

Reply:
The above solution is breaking the other unit test. Adding below condition to support both case. Will update the util function as per above comment in new task once this PR get merge. 

```
if actual_data == expected_data:
            continue
```